apps/docs/pages/guides/platform/custom-domains.mdx

```diff
@@ -18,139 +18,213 @@ Custom domains allow you to present a branded experience to your users. Custom d
 
-- Any OAuth providers (e.g., google, github)
+### Activate your domain
 
-Additionally, update the DNS configuration for `api.example.com` to once more use a CNAME record that resolves to `foobarbaz.supabase.co`.
-Finally, you can use the `activate` subcommand to reconfigure your project:
+Once you've done the necessary preparations to activate the new domain for your project, you can activate it using the [`domains activate`](/docs/reference/cli/supabase-domains-activate) CLI command.
 
 ```bash
-supabase domains activate --project-ref foobarbaz --experimental
+supabase domains activate --project-ref abcdefghijklmnopqrst
 ```
 
-## Remove a Custom Domain
+When this step completes, Supabase will serve the requests from your new domain. The Supabase project domain **continues to work** and serve requests so you do not need to rush to change client code URLs.
 
-If you have a custom domain (`api.example.com`) set up for your Supabase project (ref `foobarbaz`, with assigned endpoints at `foobarbaz.supabase.co`), and would like to go back to using the Supabase-provisioned endpoints (`foobarbaz.supabase.co`), you can use the `delete` subcommand:
+If you wish to use the new domain in client code, you can set it up like so:
 
-```bash
-supabase domains delete --project-ref foobarbaz --experimental
+```js
+import { createClient } from '@supabase/supabase-js'
+
+// Use a custom domain as the supabase URL
+const supabase = createClient('https://api.example.com', 'public-anon-key')
 ```
 
-As with the final activation stage of the process for setting up a custom domain, you'll need to update any references in your client code and OAuth providers from the custom domain to the Supabase-provisioned endpoints.
+When using [Sign in with Twitter](/docs/guides/auth/social-login/auth-twitter) make sure your frontend code is using the custom domain only!
 
-## Vanity Subdomains
+### Remove a custom domain
 
-Vanity Subdomains allow you to present a basic branded experience, compared to custom domains. They allow you to host your services at a custom subdomain on Supabase (e.g., `my-example-brand.supabase.co`) instead of the default, randomly-assigned `foobarbaz.supabase.co`.
+Removing a custom domain may cause some issues when using Supabase Auth with OAuth or SAML. You may have to reverse the changes made in the _[Prepare to activate your domain](#prepare-to-activate-your-domain)_ step above.
+
+To remove an activated custom domain you can use the [`domains delete`](/docs/reference/cli/supabase-domains-delete) CLI command.
+
+```bash
+supabase domains delete --project-ref abcdefghijklmnopqrst
+```
+
+## Vanity subdomains
+
+Vanity subdomains allow you to present a basic branded experience, compared to custom domains. They allow you to host your services at a custom subdomain on Supabase (e.g., `my-example-brand.supabase.co`) instead of the default, randomly-assigned `abcdefghijklmnopqrst.supabase.co`.
```

Comment:
```suggestion
Vanity subdomains allow you to present a basic branded experience, compared to custom domains. They allow you to host your services at a custom subdomain on Supabase (e.g., `my-example-brand.supabase.co`) instead of the default, randomly assigned `abcdefghijklmnopqrst.supabase.co`.
```